program/src/state/stake.rs

```diff
@@ -48,4 +47,14 @@ pub struct Stake {
 
 impl Stake {
     pub const LEN: usize = std::mem::size_of::<Stake>();
+
+    #[inline(always)]
+    pub fn is_initialized(&self) -> bool {
+        self.discriminator.as_slice() == Stake::SPL_DISCRIMINATOR_SLICE
+    }
+
+    #[inline(always)]
+    pub fn is_uninitialized(&self) -> bool {
+        self.discriminator.as_slice() == ArrayDiscriminator::UNINITIALIZED.as_slice()
+    }
```

Comment:
The issue with `!is_initialized()` is that it could be "weak", since you can pass another account of the same program and it will treat it as uninitialized because the discriminator don't match instead of checking that the discriminator is zeroed.

Maybe the naming is not idea, but what `is_initialized` and `is_uninitialied` are doing is checking the discriminator on the account.

Reply:
I don't even see `is_uninitialized` being used in this branch. I think it makes more sense for `InitializeConfig` because the program requires the account to come pre-allocated, while this one does not.

Reply:
Oh, yeah, good point. It won't be used as the `Config` one. I will remove it.